404.html

```diff
@@ -4,6 +4,11 @@
 <head>
     <title>adobe.com: Page not found</title>
     <script type="text/javascript">
+        if (window.location.pathname.endsWith('.html')) {
+          // redirect from .html to extensionless URL
+          window.location.replace(window.location.pathname.split('.html')[0]);
```

Comment:
This should probably be more resilient to hashes and query params, but maybe we don't have anything pointing to html anymore? One would hope.

Reply:
Resilient as in send hashes and query params along? You may be right, some paid SEO links do come with those...